### PR TITLE
Toggle OIDC basic auth header

### PR DESCRIPTION
**Pull request by taheris**
Signed-off-by: Shaun Taheri <shaun.taheri@here.com>

### CODE REVIEW

File: ota-plus-web/app/com/advancedtelematic/auth/oidc/OidcGateway.scala [hunk 1]
```diff
@@ -245,7 +254,8 @@ object OidcConfiguration {
   def load(wSClient: WSClient, configUrl: Url, issuerUrl: Url, log: Logger)(
       implicit ec: ExecutionContext
   ): Future[ProviderMetadata] = {
-    val uri = Uri(configUrl).withPath(OidcConfigurationPath)
+    val root = Uri(configUrl)
+    val uri  = root.withPath(root.path ++ OidcConfigurationPath)
```
Comment (jerrytrieu) [inline ota-plus-web/app/com/advancedtelematic/auth/oidc/OidcGateway.scala]: Is this a bugfix?

File: ota-plus-web/conf/application.conf [hunk 2]
```diff
@@ -68,9 +68,10 @@ oidc {
     jwksUri = ${oidc.fallback.issuer}".well-known/jwks.json"
     userInfoEndpoint = ${oidc.fallback.issuer}"userinfo"
   }
+  basicAuthHeader = false
```
Comment (koshelev) [inline ota-plus-web/conf/application.conf]: basic authentication is not the only possible option, so I would prefer to have `clientAuthentication` param with two options  - `PasswordBasicAuth` and `PasswordRequestBody`, default `PasswordRequestBody`

File: ota-plus-web/app/com/advancedtelematic/auth/oidc/OidcGateway.scala (hunk 1 above)
Comment (koshelev) [reply ota-plus-web/app/com/advancedtelematic/auth/oidc/OidcGateway.scala]: More like support for additional options from standard, which we need for angel delight

**Comment by taheris** [comment]
@koshelev @jerrytrieu: updated with suggestions and tested on madrid/ericsson

**Pull request merged**